### PR TITLE
Terminal input/output via websockets not keeping session alive

### PR DESCRIPTION
**Pull request by gtritchie**
The terminal websocket activity won't reset session timer, so if all you are doing is working in the terminal (but not doing a full-screen thing to make it "busy"), the session suspend timer will run out and sadness will occur.

Fixed to have output in a terminal treated as activity (at least, for a second), which will contribute to keeping the session alive.

Also put this in the Win32 implementation, just to keep symmetry with Posix code.

### CODE REVIEW

File: src/cpp/core/system/PosixChildProcess.cpp [hunk 1]
```diff
@@ -906,6 +927,7 @@ void AsyncChildProcess::poll()
    pAsyncImpl_->initTimers(options().reportHasSubprocs);
    if (pAsyncImpl_->checkChildCount())
    {
+      pAsyncImpl_->hasRecentOutput_ = false;
```
Comment (jmcphers) [inline src/cpp/core/system/PosixChildProcess.cpp]: Do we also only set the `hasRecentOutput_` flag when we're checking the child process count?

Comment (gtritchie) [inline src/cpp/core/system/PosixChildProcess.cpp]: `hasRecentOutput_` gets flipped to false every second (that's the frequency of `checkChildCount` being true, and is set to true whenever there is output seen in the terminal polling loop. The value of hasRecentOutput_ is checked (or'd with the child-process flag) here: https://github.com/rstudio/rstudio/blob/88b6ae2139cae85bcdb3f9aa03219e9285ca3305/src/cpp/session/SessionHttpMethods.cpp#L401

Comment (gtritchie) [inline src/cpp/core/system/PosixChildProcess.cpp]: If there is a long-running async child process that ISN'T a terminal (and thus doesn't do child process counting), then both these flags would stay "true", thus seen as busy and would prevent suspending.

Comment (jmcphers) [inline src/cpp/core/system/PosixChildProcess.cpp]: Appreciate the explanation, that's clear now.

**Pull request merged**